src/services/shop/shop-router.test.ts

```diff
@@ -0,0 +1,431 @@
+    it("allows user to remove an item from the cart", async () => {
+        // Add item to the cart first
+        await Models.ShopOrder.updateOne({ userId: TESTER_PROFILE.userId }, { items: ["test-item-1"], quantity: [0] });
+        await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        // Remove the item
+        await delAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+        const updatedOrder = await Models.ShopOrder.findOne({ userId: TESTER_PROFILE.userId });
+        expect(updatedOrder?.items).not.toContain(TESTER_SHOP_ITEM.itemId);
+    });
+
+    it("decreases the quantity of an item in the cart", async () => {
+        // Add the item to the cart twice
+        await Models.ShopOrder.updateOne({ userId: TESTER_PROFILE.userId }, { items: ["test-item-1"], quantity: [0] });
+        await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+        await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        // Remove the item once
+        const response = await delAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        expect(JSON.parse(response.text)).toMatchObject({
+            ...TESTER_SHOP_ORDER,
+            quantity: [1],
+        });
+
+        const updatedOrder = await Models.ShopOrder.findOne({ userId: TESTER_PROFILE.userId });
+        expect(updatedOrder?.items).toContain(TESTER_SHOP_ITEM.itemId);
+        expect(updatedOrder?.quantity[0]).toBe(1); // Item quantity should now be 1
+    });
+
+    it("removes the item completely if the quantity reaches 0", async () => {
+        // Add the item to the cart twice
+        await Models.ShopOrder.updateOne({ userId: TESTER_PROFILE.userId }, { items: ["test-item-1"], quantity: [0] });
+        await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+        await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        // Remove the item twice
+        await delAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+        await delAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        const updatedOrder = await Models.ShopOrder.findOne({ userId: TESTER_PROFILE.userId });
+        expect(updatedOrder?.items).not.toContain(TESTER_SHOP_ITEM.itemId); // The item should be completely removed
+    });
+
+    it("returns NotFound for non-existent item in cart", async () => {
+        await delAsAttendee("/shop/cart/non-existent-item").expect(StatusCode.ClientErrorNotFound);
+    });
+
+    it("returns InternalServerError when no cart exists for the user", async () => {
+        // Delete the user's cart if exists
+        await Models.ShopOrder.deleteOne({ userId: TESTER_PROFILE.userId });
+
+        // Try removing an item from a non-existent cart
+        await delAsAttendee("/shop/cart/test-item-1").expect(StatusCode.ServerErrorInternal);
```

Comment:
this shouldn't be possible - internal errors should only happen if something goes wrong, not if a user does something normal